### PR TITLE
[primTorch] Prototype tracer and elementwise unary reference opinfo class

### PR DESCRIPTION
**Pull request by mruberry**
Adds a prototype tracer with no caching support and the `ElementwiseUnaryPythonRefInfo` class. A reference for `floor` is added to test the latter, and the elementwise binary reference inputs are extended to also return noncontiguous inputs. The SampleInput transform operation has been updated to return an actual SampleInput instead of a tuple to facilitate uniform handling of (transformed) SampleInputs.

### CODE REVIEW

**Review by ngimel — APPROVED** [review]
Nice!

File: torch/_prims/utils.py [hunk 1]
```diff
@@ -3,19 +3,37 @@
...
 import torch
+from torch.fx import Node
 
 ShapeType = Union[torch.Size, List[int], Tuple[int, ...]]
 StrideType = Union[List[int], Tuple[int, ...]]
 DimsType = Union[int, List[int], Tuple[int, ...]]
 
 
-class TensorMeta(object):
+class TensorMeta_Meta(type):
```
Comment (ezyang) [inline torch/_prims/utils.py:16]: Is there any good explanation why we need a metaclass for TensorMeta?

File: torch/_prims/executor.py [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+def execute(ctx: PrimContext, *args, **kwargs):
+    """
+    Prototype ATen executor.
+
+    Just executes the context's graph.
+    """
+
+    gm = GraphModule({}, ctx.graph)
+    return gm.forward(*args, **kwargs)
```
Comment (ezyang) [inline torch/_prims/executor.py:13]: This doesn't seem to be tested anywhere?

Comment (mruberry) [reply torch/_prims/executor.py:13]: Correct - just used to support development of the prototype nvFuser Python API

File: torch/_prims/utils.py (hunk 1 above)
Comment (mruberry) [reply torch/_prims/utils.py:16]: Idk about "good" -- there's probably a better way to achieve these, but it lets us add a thread local class property

**Pull request closed without merging**